lib/devices_api/signin/commands/user_signin.ex

```diff
@@ -0,0 +1,24 @@
+defmodule DevicesApi.Signin.Commands.UserSignin do
+  @moduledoc """
+  Executes User signin by email and password
+  """
+
+  alias DevicesApi.Signin.Inputs.SigninRequestInput
+  alias DevicesApi.Signin.Steps.CheckPassword
+  alias DevicesApi.Signin.Steps.IdentifyUser
+  alias DevicesApi.Signin.Steps.TokenCreate
+  alias DevicesApi.Users.Schemas.User
+
+  @doc false
+  @spec execute(input :: SigninRequestInput.t()) ::
+          {:error, {:forbidden, String.t()}} | atom() | {:ok, String.t()}
+  def execute(%SigninRequestInput{} = input) do
+    with {:ok, %User{} = user} <- IdentifyUser.execute(input.email),
+         :ok <- CheckPassword.execute(input.password, user.password_hash) do
+      TokenCreate.execute(user)
+    else
+      {:error, _} -> {:error, {:forbidden, "Invalid Credentials!"}}
+      _ -> :error
```

Comment:
Cuidado com estes patterns aqui!!! Isso é algo importante de se acostumar de ler e não deixar passar...

Pense que, por algum motivo, `IdentifyUser.execute/1` retorne além de {:ok, user} algo como `{:ok, user, message}`, ou mesmo algo como `{:ok, %Subject{}}`. Nestes casos, você:
- não teria logs da chamada para possível debug em PROD
- o erro seria tratado silenciosamente no controller

Aqui vale a famosa filosofia do Erlang: ["let it crash"](https://medium.com/@vamsimokari/erlang-let-it-crash-philosophy-53486d2a6da). Não dê matches muito genéricos. Trate aquilo que se conhece e outros erros deixe quebrar mesmo. É preferível aqui ver um 500 e isso alertar do que ser silenciosamente ignorado.